opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -20,73 +20,80 @@
+   * @param carrier the carrier for the SpanContext state; when inject() returns, the Tracer implementation will have represented the SpanContext within `carrier`
    *
-   * Attempting to inject to a carrier that has been registered/configured to this Tracer will result in a
-   * IllegalStateException.
+   * All implementations must support, at minimum, the BuiltinFormats.
    *
-   * All implementations support at minimum the required carriers BinaryWriter and TextMapWriter.
+   * @see BuiltinFormats
    */
-  <T> void inject(Span span, T carrier);
+  <W> void inject(SpanContext spanContext, Format<?, W> format, W carrier);
 
-  /**  Returns a SpanBuilder provided
-   *    a “carrier” object from which to extract identifying information needed by the new Span instance.
-   *
-   * If the carrier object has no such span stored within it, a new Span is created.
+  /**
+   * Extract a SpanContext from a `carrier` of a given `format`, presumably after propagation across a process boundary.
    *
-   * Unless there’s an error, it returns a SpanBuilder.
-   * The Span generated from the builder can be used in the host process like any other.
+   * <p>Example:
+   * <pre>{@code
+   * Tracer tracer = ...
+   * TextMapReader httpHeaderReader = new AnHttpHeaderCarrier(httpRequest);
+   * SpanContext spanCtx = tracer.extract(BuiltinFormats.HTTP_HEADER, httpHeaderReader);
+   * tracer.buildSpan('...').withChildOf(spanCtx).start();
+   * }</pre>
    *
-   * (Note that some OpenTracing implementations consider the Spans on either side of an RPC to have the same identity,
-   * and others consider the caller to be the parent and the receiver to be the child)
+   * If the span serialized state is invalid (corrupt, wrong version, etc) inside the carrier this will result in an IllegalArgumentException.
    *
-   * Attempting to join from a carrier that has been registered/configured to this Tracer will result in a
-   * IllegalStateException.
+   * @param format the Format of the carrier. See BuiltinFormats.
+   * @param carrier the carrier for the SpanContext state
+   * @returns the SpanContext instance extracted from the carrier
    *
-   * If the span serialized state is invalid (corrupt, wrong version, etc) inside the carrier this will result in a
-   * IllegalArgumentException.
+   * All implementations must support, at minimum, the BuiltinFormats.
    *
-   * All implementations support at minimum the required carriers BinaryReader and TextMapReader.
+   * @see BuiltinFormats
    */
-  <T> SpanBuilder join(T carrier);
+  <R> SpanContext extract(Format<R, ?> format, R carrier);
 
 
   interface SpanBuilder {
 
-      /** Specify the operationName.
-       *
-       * If the operationName has already been set (implicitly or explicitly) an IllegalStateException will be thrown.
+      /**
+       * A shorthand for withReference(Reference.childOf(parent)).
        */
-      SpanBuilder withOperationName(String operationName);
```

Comment:
still there, of course, when creating the builder... but another nice thing about the move towards SpanContext (and extract instead of join) is that you basically never need to create a span when you don't already know the operation name.